cmd/create.go

```diff
@@ -128,9 +130,26 @@ var createCmd = &cobra.Command{
 		}
 
 		project.AbsolutePath = currentWorkingDir
-
-		// This calls the templates
+		spinner := tea.NewProgram(spinner.InitialModelNew())
+		// add synchronization to wait for spinner to finish
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			// only run the spinner if the command is interactive
+			if isInteractive {
+				if _, err := spinner.Run(); err != nil {
+					cobra.CheckErr(err)
+				}
+			}
+		}()
 		err = project.CreateMainFile()
+		// once the create is done, stop the spinner
+		if isInteractive {
+			spinner.Quit()
+		}
+		// wait for the spinner to finish
+		wg.Wait()
```

Comment:
Actually, that's an interesting approach. But I want to get something clear here; correct me if I'm wrong. First, we initialize the spinner model, then add one to the wait group indicating that we have one go-rotune we need to wait for. After that, we start the spinner in a goroutine, and then we begin creating the main file. Now, the point is, does the goroutine actually get marked as done after using spinner.Quit()? and after that wg.Wait() unblock the main routine? Or am I missing something here? Because I didn't totally understand how the goroutine is going to get marked as done after we sta

Reply:
I'm not sure what you mean by marked as done.

Here is what is going to happen (assuming this is in interactive mode):

- We initiate a wait group and add 1 to the wait for the spinner
- We start the spinner - the spinner function is a blocking function that does not actually finish running until you call `Quit()` on it.
- We then call `Quit()` on it, which ends the run previously started in the above go routine.
- Once that function exits, the defer is called for the `wg.Done()`
- Since we were only waiting for 1 item to finish, `wg.Wait()` considers everything is done (since the go routine existed) and the program resumes it's normal execution